### ISSUE TITLE
Is it possible to size the feature images the same as the posts?

### ISSUE THREAD

**Issue by paule96** [GetPublii/Publii#631]
Hi Publii team,

befor I open a PR, I want ask if I can change [this line ](https://github.com/GetPublii/Publii/blob/57e35b86ad0ffc1ec545b86d9259739b637b47bd/app/default-files/default-themes/simple/assets/css/style.css#L1233) from:

```css
  max-width: calc(42rem + 20%);
```

to

```css
  max-width: 42rem;
```

The reason why I want this because it looks realy wired if you enabled feature pictures in your feed. The picture is much bigger then the text to it. Here a sample:

![image](https://user-images.githubusercontent.com/11291885/70550607-4bcc6a00-1b76-11ea-8621-755741d47fc2.png)

After the proposed change it looks like this:

![image](https://user-images.githubusercontent.com/11291885/70550713-78808180-1b76-11ea-97e1-2aea64eb6f3b.png)

**Comment by Dorson** [comment]
The issue is that the default CSS is designed to handle page padding and margin sizes individually for all elements.  It's creates bloat and issues in the long term.

**Use minimal global settings** for those margins. Individual margins are only needed in edge cases, where they could be set individually to overwrite the global setting.

For example document-wide, softly-shaded page borders like in printed pages:

**html {padding:0em 1em;font-size:115%; background:#F9FDFE;}**

Not sure if styling the HTML tag is valid code, but I use that for ages without issues. works great for mobile and tablets too.

Set the <body> tag internal side padding to 1-2 em (text lines) for all content inside, instead of setting the margins and padding for hundreds of elements later on. Better once, than 100x times for every element and div box.

body {**margin: 0px;  padding: 0px 0.7em 3px 3px;  width:99%;  display:block;**
  background: #FCFDFE;
  font: 1em/1.5em Verdana , sans-serif ;
  color: #fff;
}

The width 99% will avoid overflow on mobile devices, if the body content size is rendered wrongly. Could be also max-width. Does cut off ghost horizontal scroll bars perfectly too.

Then all the content has equal distance from the sides, no need to set margins for anything individually, which paints pages faster.

Then the super tick of max width of all possible over-sized elements (works on mobile and tablets too !):
set CSS max-width as:

 **img, iframe, input, element, javascript, embed, body, html,table, div, {max-width:100%;}**

At least set that for images, the iframe and javascript for ads or code widgets that might be loaded from external sources, and are not on the page itself.

Then maybe define the **default minimal margin** for images, text fields, tables, and similar elements that might exists inside of body. At least for images.

**img, table, iframe, textarea {margin: 0.5em;}**

The individual margin can still be over-written later on in individual cases, but there will be no need to set it for every elements at all, if the default margin works.


We could also set the global **body * {max-width:100%;}** to force all elements to be limited to screen width size (mobile and tablet !) Would work, if the **body width is already set at 99% of screen size,** which makes all possible seizes to be limited to the 99% of screen width, and avoids overflow on mobile screens.

**In short:** There is absolutely no need to set sizes and margin for individual elements every time, if we can define minimal global defaults, and then set edge cases like text and input fields, or menu elements right when we use them.


The default CSS could be much leaner, if we avoid CSS classes, and define global minimal settings for basic elements, and rewrite them in edge cases.

**Comment by bobmitro** [comment]
What are you talking about? Where do you see the elements with 100x of the same CSS rules? 
Your idea is suitable for specific cases, for example, padding: 0px 0.7em 3px 3px - it will look ok when you want to have padding for the whole page, but what about if you want to use the full 100% hero section?  For me, it is much better to use a container with padding than I can apply to the sections that need it. 

"There is absolutely no need to set sizes and margin for individual elements every time" - I agree with you entirely, but in the above case, the intention was to take pictures outside of content.  

Every situation requires a different approach, so this is why such a conversation does not make much sense to me.  

What prevents you from creating your own theme according to your idea? Just do it, the community will benefit.

BTW/ guys, this Github repo is dedicated to the app, not the themes. If you want to get help from the Publi community, please use the forum (https://getpublii.com/forum/).

**Comment by Dorson** [comment]
> What are you talking about? Where do you see the elements with 100x of the same CSS rules?

It's all related to themes. Could be resolved in a different CSS theme. Not an issue of the app.

I am talking about the CSS classes that set all the settings individually in the theme that the guy above (paule96) has an issue with:

in this theme:
https://github.com/GetPublii/Publii/blob/57e35b86ad0ffc1ec545b86d9259739b637b47bd/app/default-files/default-themes/simple/assets/css/style.css#L1233


margin-top: setting found 34 times :-)
padding:  setting found 58 times :-)

background: setting found 57 times :-)

In the object style coding we could just treat the CSS padding setting as a global object variable or value. It's reverse psychology, where the setting is the global rule object and the html element is the variable to be connected to the global CSS setting.

**In the mentioned default theme** it's used repetitively as a function for every other CSS class.

**The better object-style rule would be:**

bla1, bla2, bla2, img, p, input, bla5, bla5 {padding: 3px 3px 0.5em 3px}

We could set this margin **only once or 10 times,** instead of 58 times :-)

Such global CSS rules work better if they are set at the top of the CSS file, and later on overwritten, when needed.

In any case it's a CSS principles issue. One could create the minimal CSS theme with the principles and examples that I made above.

Not an app issue in general. I hope it helps !

**Comment by paule96** [comment]
@bobmitro but the default theme is placed in this repo so I think this is the right spot to discuss. Your are right for the hero the 100% is good and okay. But what I'm showing above is not the hero port. The screenshot are from the "post list". And if you enable "feature pictures" in the "post list" you get this wired result. I like the default theme for all the rest but this is a little pain. Also I find out that the default theme need some adjustments for mobile sites. The current changes I must make to get my site working on mobile and on desktop are to followings:

```css
/*
 * Put your custom CSS code here
 */
main {
  /*This is for some spacing in mobil for the header. This is currently not stabel on all sizes*/
  margin-top: 4.4rem;
  padding-top: 1rem;
  background: #FFF;
}

header.top {
  /*This is for some spacing in mobil for the header. This is currently not stabel on all sizes*/
  background: #538a3c;
}

.feed__image{
  /* This is the fix to show the images in a posts feed in the right size.*/
  max-width: 42rem;
}

.post__entry {
  /* This fix tables on mobile devices. Happy scroll 🧙‍*/
  overflow-x: overlay;
}

.hero__content{  
  /*This defines a little space between the navigation and the text in the hero section. To you can see all of the text.*/
  margin-top: 1rem;
}

@media only screen and (max-width: 768px) {
  /* This makes the h1 much smaller on mobil devices because they shouldn't leave the hero box.*/
  h1{
    font-size: large
  }
}
```

This all is work in progress and I think I will change something more soon. So maybe we can rename or reopen the issue to fix some small problems in the default theme. :)

> Disclaimer: I'm new to Publii and I like it much. 👍 

Thanks to @asdeadasad0do to find the table bug 😺

**Comment by bobmitro** [comment]
@Dorson  This way, by collecting all elements/classes together you lose the possibility of easy editing and individual styling of the elements is very complicated; good for simple, "blank" theme, not for more advanced.
@paule96 For me it doe not look weird; if you think differently you can easily change the way it is styled by modifying or adding your own CSS rules. 

Once again, when you want to get help related to themes, please use the forum.
